docs/content/testing/writing-a-new-test.md

```diff
@@ -0,0 +1,344 @@
+            Assert.AreEqual(4, MyClass.AddTwoNumbers(3, 1));
+            Assert.AreEqual(7, MyClass.AddTwoNumbers(2, 5));
+        }
+    }
+```
+
+For the instance functionality, you can add tests similarly:
+
+```cs
+    [TestFixture]
+    public class MyClassInstanceTests
+    {
+        private MyClass myClassInstance;
+
+        [SetUp]
+        public void SetUp()
+        {
+            myClassInstance = new MyClass();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            myClassInstance.Dispose();
+        }
+
+        [Test]
+        public void InstanceValue_is_0_by_default()
+        {
+            Assert.AreEqual(0, myClassInstance.InstanceValue);
+        }
+
+        [Test]
+        public void DoInstanceOperation_increases_InstanceValue_by_1()
+        {
+            myClassInstance.DoInstanceOperation();
+
+            Assert.AreEqual(1, myClassInstance.InstanceValue);
+
+            myClassInstance.DoInstanceOperation();
+
+            Assert.AreEqual(2, myClassInstance.InstanceValue);
+        }
+
+        [Test]
+        public void DoSomethingElse_increses_InstanceValue_by_5()
+        {
+            myClassInstance.DoSomethingElse();
+
+            Assert.AreEqual(5, myClassInstance.InstanceValue);
+
+            myClassInstance.DoSomethingElse();
+
+            Assert.AreEqual(10, myClassInstance.InstanceValue);
+        }
+    }
+```
+
+### Internal variables and methods
+
+The tests above will not compile.
+
+This is because the test class is in another assembly than the implementation
+ class, and the tests are looking at internal fields.
+
+There is a solution. In the `MyClass.cs` file, you can make its internals
+ visible to the test assembly:
+
+```
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Improbable.Gdk.Core.EditmodeTests")]
+```
+
+### Running the new tests
+
+In the Test Runner window of Unity, you can find the test under:
+- Improbable.Gdk.Core.EditmodeTests.dll (this matches the folder name
+ `Gdk/Core/Tests/Editmode`)
+
+And you can follow the namespace and fixture name in the hierarchy:
+
+- `Improbable` > `Gdk` > `Core` > `EditmodeTests` > `Utility` > `MyClassStaticTests`,
+- `Improbable` > `Gdk` > `Core` > `EditmodeTests` > `Utility` > `MyClassInstanceTests`.
+
+Double-click the fixture name to run all tests within that fixture.
+
+## Creating a new test folder and assembly
+
+Let's say you want to add tests for the `/workers/unity/Assets/Gdk/Legacy`
+ folder within the Unity project's assets.
+
+Create the directory:
+
+`/workers/unity/Assets/Gdk/Legacy/Tests/Editmode`
+
+Right click the directory in the Project window in Unity, and select
+ `Create` > `Assembly Definition`.
+
+Name this file to match:
+
+`Improbable.Gdk.Legacy.EditmodeTests`.
+
+Select this file, and change the `name` property in the inspector to match the
+ filename:
+
+`Improbable.Gdk.Legacy.EditmodeTests`.
+
+Hit the "Apply" button.
```

Comment:
same as above for all the "some text", should we always use ".." or `...`?

Reply:
Aye